mlir/lib/Target/Cpp/TranslateToCpp.cpp

```diff
@@ -295,10 +304,10 @@ static bool shouldBeInlined(ExpressionOp expressionOp) {
 
   Operation *user = *result.getUsers().begin();
 
-  // Do not inline expressions used by subscript operations, since the
-  // way the subscript operation translation is implemented requires that
-  // variables be materialized.
-  if (isa<emitc::SubscriptOp>(user))
+  // Do not inline expressions used by operations with deferred emission, since
+  // the way their translation is implemented requires that variables be
+  // materialized.
```

Comment:
Maybe rephrase to
```suggestion
  // the way their translation is implemented requires that variables are
  // materialized.
```
or
```suggestion
  // the way their translation is implemented requires to materialize
  // variables.
```
or
```suggestion
  // their translation requires the materialization of variables.
```